src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -51,46 +52,121 @@ internal sealed class TupleTypeSymbol : NamedTypeSymbol, ITupleTypeSymbol
-                    }
+                for (int i = 0; i < elementTypes.Length; i++)
+                {
+                    var wellKnownTypeMember = EnumTypeMember(elementTypes.Length, i);
+                    var underlyingField = (FieldSymbol)Binder.GetWellKnownTypeMember(binder.Compilation, wellKnownTypeMember, diagnostics, syntax: syntax);
+                    var field = new TupleFieldSymbol(elementNames.IsDefaultOrEmpty ? UnderlyingMemberName(i) : elementNames[i],
+                                               this,
+                                               elementTypes[i],
+                                               underlyingField?.AsMember(underlyingType));
+                    fieldSymbols.Add(field);
+                }
+
+                fields = ImmutableArray.Create(fieldSymbols.ToImmutableAndFree(), 0, elementTypes.Length);
+            }
+            else
+            {
+                fields = ImmutableArray<TupleFieldSymbol>.Empty;
 
+                // PROTOTYPE: VS if this eventually still stays reachable, need to make some error type symbol
             }
 
             return underlyingType;
         }
 
+        /// <summary>
+        /// Find the well-known ValueTuple type of a given arity.
+        /// For example, for arity=2:
+        /// returns WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2
+        /// </summary>
+        private static WellKnownType EnumType(int arity)
+        {
+            return tupleTypes[arity - 1];
+        }
+
+        private static WellKnownType[] tupleTypes = { WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1,
+                                                        WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2,
+                                                        WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3,
+                                                        WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4,
+                                                        WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5,
+                                                        WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6,
+                                                        WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7 };
+
+        /// <summary>
+        /// Find the well-known members to the ValueTuple type of a given arity and index/position.
+        /// For example, for arity=3 and position=0:
+        /// returns WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3__Item1
+        /// </summary>
+        private static WellKnownMember EnumTypeMember(int arity, int position)
+        {
+            return tupleMembers[arity - 1][position];
+        }
+
+        private static WellKnownMember[] tuple1Members = { WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1__Item1 };
```

Comment:
@AlekseyTs, you mean as opposed to inlining everything? I had it that way first, but thought breaking them out was more readable.